lighthouse-core/lib/tracehouse/trace-processor.js

```diff
@@ -82,11 +82,12 @@ class TraceProcessor {
    * Returns true if the event is a navigation start event of a document whose URL seems valid.
    *
    * @param {LH.TraceEvent} event
+   * @return {boolean}
    */
   static _isNavigationStartOfInterest(event) {
-    return event.name === 'navigationStart' &&
-      (!event.args.data || !event.args.data.documentLoaderURL ||
-        ACCEPTABLE_NAVIGATION_URL_REGEX.test(event.args.data.documentLoaderURL));
+    if (event.name !== 'navigationStart') return false;
+    if (!event.args.data?.documentLoaderURL) return false;
```

Comment:
maybe some variation on
```suggestion
    // COMPAT: support pre-m67 test traces before `args.data` added to all navStart events.
    // TODO: remove next line when old test traces (progressive-app-m60.json) are updated.
    if (!event.args.data) return true;
    if (!event.args.data?.documentLoaderURL) return false;
```
?

I'm 100% good with updating well past m60 for a test trace, but we could fix the test failure to unblock CI and then update the tests? Maybe it would be a trivial update, but we'd want to make sure it exercises the same execution paths, and it's currently used in at least 227 tests, probably more that just aren't currently failing with `NO_NAVSTART`.

Reply:
for future reference, CL from @deepanjanroy that added `args.data` to all navStart events: https://chromium-review.googlesource.com/975986

Reply:
Ope I just thought that the unit failures were all https://github.com/GoogleChrome/lighthouse/pull/13846. Didn't think to check.

Reply:
There were some dest traces with `event.args.data` but no `documentLoaderURL`. I updated this to check if `documentLoaderURL` is specifically `undefined`.

Reply:
haha, ok then, so when we need to update the trace tests just removing that one line will be where to start